modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -70,7 +70,12 @@ func (k Keeper) OnChanOpenInit(
 			return "", sdkerrors.Wrapf(icatypes.ErrActiveChannelAlreadySet, "existing active channel %s for portID %s is already OPEN", activeChannelID, portID)
 		}
 
-		if !icatypes.IsPreviousMetadataEqual(channel.Version, metadata) {
+		appVersion, found := k.GetAppVersion(ctx, portID, activeChannelID)
+		if !found {
+			panic(fmt.Sprintf("active channel mapping set for %s, but channel does not exist in channel store", activeChannelID))
```

Comment:
looks like this panic introduces multiple conventions in the function (error return + panic) does it make more sense to return an error instead?

Reply:
Do we need to panic?

Reply:
As mentioned earlier in person I just followed essentially the same used above [here](https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/controller/keeper/handshake.go#L64-L67) - as in some places in ibc-go code we just `panic` because it's essentially unreachable code.

Reply:
maybe we could add a comment indicating as such? The message we panic with seems like it would be a regular error that we could run into.